bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/services/impl/DynamicDeckServiceImpl.java

```
@@ -0,0 +1,224 @@
+
+        if (StringUtils.isEmpty(inddTemplatePath) || StringUtils.isEmpty(annotatedXmlPath)) {
+            throw new DynamicDeckDynamoException("Supplied INDD template folder path doesn't contain InDesign Template file OR template XML file : " + templateFolderPath);
+        }
+
+        StringBuilder idspScriptArgs = new StringBuilder();
+        List<String> inddImageList = new ArrayList<>();
+
+        Resource destinationFolderResource = resourceResolver.getResource(destinationFolderPath);
+        if (null == destinationFolderResource) {
+            throw new DynamicDeckDynamoException("Destination folder resource is null, hence exiting the deck generation process.");
+        }
+
+
+        Asset damAsset = DynamicDeckUtils.createUniqueAsset(destinationFolderResource, JcrUtil.createValidName(deckName) + DynamicDeckDynamoConstants.DOT + DynamicDeckDynamoConstants.INDD_EXTENSION, resourceResolver);
+        if (damAsset == null) {
+            throw new DynamicDeckDynamoException("Dynamic Deck document not created at the destination : " + destinationFolderResource.getPath());
+        }
+
+        InputStream templateXmlInputStream = DynamicDeckUtils.getAssetInputStreamByPath(resourceResolver, annotatedXmlPath);
+        if (null == templateXmlInputStream) {
+            throw new DynamicDeckDynamoException("InDesign Template XML is null, hence exiting the deck generation process");
+        }
+
+        List<XMLResourceIterator> assetItrList = new ArrayList<>();
+        assetItrList.add(new XMLResourceIterator(DynamicDeckDynamoConstants.XML_SECTION_TYPE_GENERIC, assetResourceList.listIterator()));
+
+        String processedXmlPath = xmlGeneratorService.generateInddXML(templateXmlInputStream, assetItrList,
+                resourceResolver.getResource(masterAssetPath), damAsset.adaptTo(Resource.class), resourceResolver, inddImageList);
```

a lot of these methods might silently return ```null```, you should validate them first (or switch the types of these parameters to Resource might help as well).